src/Nethermind/Nethermind.Consensus.AuRa/Validators/ContractBasedValidator.cs

```diff
@@ -164,7 +158,7 @@ private PendingValidators TryGetInitChangeFromPastBlocks(Keccak blockHash)
                 {
                     if (Validators.SequenceEqual(potentialValidators))
                     {
-                        break;
+                        break; // TODO: why this?
```

Comment:
I think the idea was that they are already finalized, might be wrong though

Reply:
I think it is wrong tbh

Reply:
what is wrong and why?

Reply:
in cases where `InitChange` events are emitted with the `pendingValidators` being equal to `currentValidators` this will not pick up the `InitChange` event, which will result in not applying `finalizeChanges`, hence creating a fork.

Reply:
You might be right, but we would have to full sync all networks without it to be sure it doesn't break anything

Reply:
Could you remove it and test in in separate PR?

Reply:
you mean remove the TODO comment?

Reply:
No the if :)